code/datums/components/crafting/crafting.dm

```diff
@@ -451,3 +451,7 @@
 	if(!learned_recipes)
 		learned_recipes = list()
 	learned_recipes |= R
+
+/datum/mind/proc/forget_crafting_recipe(R)
+	if(learned_recipes && (R in learned_recipes))
```

Comment:
```suggestion
	if(learned_recipes)
```

Removal through `-=` just won't happen (without any problem) when `R` isn't in `learned_recipes`. `in` operator generally isn't needed.